code/modules/mob/mob.dm

```diff
@@ -1137,18 +1137,84 @@
 	if(client.mouse_override_icon)
 		client.mouse_pointer_icon = client.mouse_override_icon
 
-
-///This mob is abile to read books
+/// Is this mob able to read and write
 /mob/proc/is_literate()
 	return FALSE
-///Can this mob read (is literate and not blind)
-/mob/proc/can_read(obj/O)
+
+/// Is this mob affected by nearsight
+/mob/proc/is_nearsighted()
+	return FALSE
+
+/* Can this mob see in the dark
+ *
+ * This checks all traits, glasses, and robotic eyeball implants to see if the mob can see in the dark
+ * this does NOT check if the mob is missing it's eyeballs. Also see_in_dark is a BYOND mob var (that defaults to 2)
+**/
+/mob/proc/has_nightvision()
+	return see_in_dark >= NIGHTVISION_FOV_RANGE
+
+/* Checks if there is enough light where the mob is located
+ *
+ * Args:
+ *  light_amount (optional) - A decimal amount between 1.0 through 0.0 (default is 0.2)
+**/
+/mob/proc/has_light_nearby(light_amount = LIGHTING_TILE_IS_DARK)
+	var/turf/mob_location = get_turf(src)
+	return mob_location.get_lumcount() > light_amount
```

Comment:
any reason why you cant just check the turfs luminosity value?

Reply:
This is the code to check lumcount:

https://github.com/tgstation/tgstation/blob/868b4cb316c41d606eb8e15b9971fd411fdefa3c/code/modules/lighting/lighting_turf.dm#L19-L46

Maybe I should use this one instead?

https://github.com/tgstation/tgstation/blob/868b4cb316c41d606eb8e15b9971fd411fdefa3c/code/modules/lighting/lighting_turf.dm#L48-L56

Reply:
![Discord_i9Fm3Jblls](https://user-images.githubusercontent.com/5195984/161665375-ce95527e-c379-4f33-b6be-92688082142e.png)

After talking with MSO, I'm going to keep `get_lumcount` since `is_softly_lit` is pretty much a deprecated function.